service/history/queues/executable_test.go

```diff
@@ -297,6 +303,34 @@ func (s *executableSuite) TestExecuteHandleErr_Corrupted() {
 	s.NoError(executable.HandleErr(err))
 }
 
+func (s *executableSuite) TestExecute_DropsInternalErrors_WhenEnabled() {
+	executable := s.newTestExecutable(func(p *params) {
+		p.dropInternalErrors = func() bool { return true }
+	})
+
+	s.mockExecutor.EXPECT().Execute(gomock.Any(), executable).DoAndReturn(
+		func(_ context.Context, _ Executable) ([]metrics.Tag, bool, error) {
+			panic(serviceerror.NewInternal("injected error"))
+		},
+	)
+
+	s.NoError(executable.HandleErr(executable.Execute()))
```

Comment:
I would introduce variable for executable.HandleErr result. This way action will be separated from assertion.